opencensus/stats/measurement_map.py

```diff
@@ -75,11 +90,27 @@ def measure_put_attachment(self, key, value):
 
         self._attachments[key] = value
 
-    def record(self, tag_map_tags=execution_context.get_current_tag_map()):
+    def record(self, tag_map_tags=None):
         """records all the measures at the same time with a tag_map.
         tag_map could either be explicitly passed to the method, or implicitly
         read from current execution context.
         """
+        if tag_map_tags is None:
+            tag_map_tags = execution_context.get_current_tag_map()
+        if self._invalid:
+            logger.warning("Measurement map has included negative value "
+                           "measurements, refusing to record")
+            return
+        for measure, value in self.measurement_map.items():
+            if value < 0:
+                self._invalid = True
```

Comment:
I don't have a strong preference, but this is intentional since I don't think it makes sense to raise until the user tries to record.

Reply:
> I don't think it makes sense to raise until the user tries to record.

Agreed. But my point was if you set ```self._invalid = True``` in the put method, during record time you just have to check ```if (self._invalid == True)``` instead of iterating on map.